presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregationsDescriptor.java

```diff
@@ -78,37 +124,76 @@ public StatisticAggregationsDescriptor(
                 map(this.getColumnStatistics(), mapper));
     }
 
-    private static <K1, K2, V> Map<K2, V> map(Map<K1, V> input, Function<K1, K2> mapper)
+    private static <K, V1, V2> Map<K, V2> map(Map<K, V1> input, Function<V1, V2> mapper)
     {
         return input.entrySet()
                 .stream()
-                .collect(toImmutableMap(entry -> mapper.apply(entry.getKey()), Map.Entry::getValue));
+                .collect(toImmutableMap(Map.Entry::getKey, entry -> mapper.apply(entry.getValue())));
     }
 
     public static class Builder<T>
     {
-        private final ImmutableMap.Builder<T, String> grouping = ImmutableMap.builder();
-        private final ImmutableMap.Builder<T, TableStatisticType> tableStatistics = ImmutableMap.builder();
-        private final ImmutableMap.Builder<T, ColumnStatisticMetadata> columnStatistics = ImmutableMap.builder();
+        private final ImmutableMap.Builder<String, T> grouping = ImmutableMap.builder();
+        private final ImmutableMap.Builder<TableStatisticType, T> tableStatistics = ImmutableMap.builder();
+        private final ImmutableMap.Builder<ColumnStatisticMetadata, T> columnStatistics = ImmutableMap.builder();
 
-        public void addGrouping(T key, String column)
+        public void addGrouping(String column, T key)
         {
-            grouping.put(key, column);
+            grouping.put(column, key);
         }
 
-        public void addTableStatistic(T key, TableStatisticType type)
+        public void addTableStatistic(TableStatisticType type, T key)
         {
-            tableStatistics.put(key, type);
+            tableStatistics.put(type, key);
         }
 
-        public void addColumnStatistic(T key, ColumnStatisticMetadata statisticMetadata)
+        public void addColumnStatistic(ColumnStatisticMetadata statisticMetadata, T key)
         {
-            columnStatistics.put(key, statisticMetadata);
+            columnStatistics.put(statisticMetadata, key);
         }
 
         public StatisticAggregationsDescriptor<T> build()
         {
             return new StatisticAggregationsDescriptor<>(grouping.build(), tableStatistics.build(), columnStatistics.build());
         }
     }
+
+    public static class ColumnStatisticMetadataKeySerializer
+            extends JsonSerializer<ColumnStatisticMetadata>
+    {
+        @Override
+        public void serialize(ColumnStatisticMetadata value, JsonGenerator gen, SerializerProvider serializers)
+                throws IOException
+        {
+            verify(value != null, "value is null");
+            gen.writeFieldName(serialize(value));
+        }
+
+        @VisibleForTesting
+        public static String serialize(ColumnStatisticMetadata value)
+        {
+            return value.getStatisticType().name() + ":" + value.getColumnName();
```

Comment:
What does "column name" refer to? The plan IR shouldn't have any references to column names, but should be using column handles, instead.